File: apps/mybookkeeper/frontend/src/shared/types/utility/utility-plan-list-response.ts
import type { UtilityPlanSummary } from "@/shared/types/utility/utility-plan-summary";

/**
 * Paginated list response for utility plans.
 *
 * Mirrors ``schemas/properties/utility_plan_list_response.py``.
 */
export interface UtilityPlanListResponse {
  items: UtilityPlanSummary[];
  total: number;
  has_more: boolean;
}

File: apps/mybookkeeper/backend/app/schemas/properties/utility_plan_list_response.py
"""Paginated list response for utility plans."""
from __future__ import annotations

from platform_shared.schemas.pagination import ListResponse

from app.schemas.properties.utility_plan_summary import UtilityPlanSummary


class UtilityPlanListResponse(ListResponse[UtilityPlanSummary]):
    pass
